### ISSUE TITLE
Pulses: An error occurred when displaying this card

### ISSUE THREAD

**Issue by pierreavizou** [metabase/metabase#9648]
- Your databases: Postgres in this case
- Metabase version: 0.32.1
- Metabase hosting environment: Docker on AWS
- Metabase internal database: H2

When a pulse is created from a question and set to run periodically, the card does not render properly and shows an error instead of displaying the actual data when it is sent to slack/email.

Source question is in the table output format and has 74 rows. 

Strangely enough, I have not yet seen the bug happen when I manually click on "send to Slack now". It seems to mainly happen when the pulse is triggered on schedule. 

*Repeatable steps to reproduce the issue*

- Create a question that queries data from a DB (tested with Postgres and a native SQL query). 
- Create a pulse from this card
- Schedule the pulse to be sent to slack at the desired time
- The card should render an error instead of actual data when sent to Slack

See error below: 
<img width="484" alt="Capture d’écran 2019-04-01 à 12 25 47" src="https://user-images.githubusercontent.com/7101583/55321181-4c562200-5479-11e9-8ab2-f0920bc773f5.png">

Below are the server logs related to the issue:

```04-01 12:23:28 DEBUG middleware.log :: GET /api/pulse/2 200 (16 ms) (8 DB calls). Jetty threads: 8/50 (4 busy, 2 idle, 0 queued) (87 total active threads)
04-01 12:23:29 ERROR sql-jdbc.execute :: Echec du réglage du fuseau horaire :
 PSQLException:
 Message: Une erreur d'entrée/sortie a eu lieu lors d'envoi vers le serveur.
 SQLState: 08006
 Error Code: 0

PSQLException:
 Message: Une erreur d'entrée/sortie a eu lieu lors d'envoi vers le serveur.
 SQLState: 08006
 Error Code: 0
04-01 12:23:29 ERROR sql-jdbc.execute :: nil
04-01 12:23:29 WARN metabase.query-processor :: {:status :failed,
 :class java.lang.Exception,
 :error "Une erreur d'entrée/sortie a eu lieu lors d'envoi vers le serveur.",
 :stacktrace
 ("--> driver.sql_jdbc.execute$do_with_try_catch.invokeStatic(execute.clj:222)"
  "driver.sql_jdbc.execute$do_with_try_catch.invoke(execute.clj:213)"
  "driver.sql_jdbc.execute$execute_query.invokeStatic(execute.clj:283)"
  "driver.sql_jdbc.execute$execute_query.invoke(execute.clj:277)"
  "driver.sql_jdbc$fn__62284.invokeStatic(sql_jdbc.clj:48)"
  "driver.sql_jdbc$fn__62284.invoke(sql_jdbc.clj:47)"
  "query_processor$fn__39538$execute_query__39543$fn__39544.invoke(query_processor.clj:67)"
  "query_processor$fn__39538$execute_query__39543.invoke(query_processor.clj:61)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__33899.invoke(mbql_to_native.clj:36)"
  "query_processor.middleware.annotate$result_rows_maps__GT_vectors$fn__32620.invoke(annotate.clj:368)"
  "query_processor.middleware.annotate$add_column_info$fn__32525.invoke(annotate.clj:303)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__33362.invoke(cumulative_aggregations.clj:57)"
  "query_processor.middleware.limit$limit$fn__33855.invoke(limit.clj:19)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__39405.invoke(results_metadata.clj:86)"
  "query_processor.middleware.format_rows$format_rows$fn__33843.invoke(format_rows.clj:26)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__31677.invoke(add_dimension_projections.clj:234)"
  "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__32071.invoke(add_row_count_and_status.clj:15)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__33604.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__36375.invoke(resolve_driver.clj:15)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__32800$fn__32801.invoke(bind_effective_timezone.clj:9)"
  "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
  "util.date$call_with_effective_timezone.invoke(date.clj:77)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__32800.invoke(bind_effective_timezone.clj:8)"
  "query_processor.middleware.store$initialize_store$fn__39430$fn__39431.invoke(store.clj:11)"
  "query_processor.store$do_with_new_store.invokeStatic(store.clj:35)"
  "query_processor.store$do_with_new_store.invoke(store.clj:31)"
  "query_processor.middleware.store$initialize_store$fn__39430.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__33222.invoke(cache.clj:127)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__33270.invoke(catch_exceptions.clj:63)"
  "query_processor$process_query.invokeStatic(query_processor.clj:232)"
  "query_processor$process_query.invoke(query_processor.clj:228)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:341)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:335)"
  "query_processor$fn__39618$process_query_and_save_execution_BANG___39623$fn__39624.invoke(query_processor.clj:371)"
  "query_processor$fn__39618$process_query_and_save_execution_BANG___39623.invoke(query_processor.clj:357)"
  "api.pulse$pulse_card_query_results.invokeStatic(pulse.clj:138)"
  "api.pulse$pulse_card_query_results.invoke(pulse.clj:137)"
  "api.pulse$fn__57638.invokeStatic(pulse.clj:159)"
  "api.pulse$fn__57638.invoke(pulse.clj:155)"
  "middleware.auth$enforce_authentication$fn__59463.invoke(auth.clj:14)"
  "routes$fn__60988$fn__60989.doInvoke(routes.clj:113)"
  "middleware.exceptions$catch_uncaught_exceptions$fn__59554.invoke(exceptions.clj:104)"
  "middleware.exceptions$catch_api_exceptions$fn__59551.invoke(exceptions.clj:92)"
  "middleware.log$log_api_call$fn__60864$fn__60865.invoke(log.clj:73)"
  "middleware.log$log_api_call$fn__60864.invoke(log.clj:66)"
  "middleware.security$add_security_headers$fn__59517.invoke(security.clj:102)"
  "middleware.json$wrap_json_body$fn__60687.invoke(json.clj:61)"
  "middleware.json$wrap_streamed_json_response$fn__60705.invoke(json.clj:120)"
  "middleware.session$bind_current_user$fn__58204$fn__58205.invoke(session.clj:159)"
  "middleware.session$do_with_current_user.invokeStatic(session.clj:142)"
  "middleware.session$do_with_current_user.invoke(session.clj:136)"
  "middleware.session$bind_current_user$fn__58204.invoke(session.clj:158)"
  "middleware.session$wrap_current_user_id$fn__58193.invoke(session.clj:127)"
  "middleware.session$wrap_session_id$fn__58178.invoke(session.clj:89)"
  "middleware.auth$wrap_api_key$fn__59471.invoke(auth.clj:27)"
  "middleware.misc$maybe_set_site_url$fn__60931.invoke(misc.clj:54)"
  "middleware.misc$bind_user_locale$fn__60934.invoke(misc.clj:70)"
  "middleware.misc$add_content_type$fn__60921.invoke(misc.clj:29)"
  "middleware.misc$wrap_gzip$fn__60949.invoke(misc.clj:97)"),
 :query
 {:type :native,
  :native
  {:query REDACTED FOR SECURITY REASONS},
  :info
  {:executed-by 1,
   :context :pulse,
   :card-id 117,
   :query-hash [3, -12, -5, 16, 109, 81, -119, -67, -111, -100, -76, -45, -44, -72, -20, 67, 29, 40, 7, 66, 58, 31, 5, 11, -32, 120, 16, -39, 14, -44, -110, -13],
   :query-type "native"}}}

04-01 12:23:29 WARN metabase.query-processor :: Query failure
04-01 12:23:29 ERROR pulse.render :: Erreur au rendu de la carte de Pulse clojure.lang.ExceptionInfo Questions en erreur : Une erreur d'entrée/sortie a eu lieu lors d'envoi vers le serveur.
 ("--> pulse.render$fn__27057$render_pulse_card_body__27062$fn__27066.invoke(render.clj:778)"
 "pulse.render$fn__27057$render_pulse_card_body__27062.invoke(render.clj:773)"
 "pulse.render$fn__27090$render_pulse_card__27095$fn__27096.invoke(render.clj:800)"
 "pulse.render$fn__27090$render_pulse_card__27095.invoke(render.clj:796)"
 "pulse.render$render_pulse_card_for_display.invokeStatic(render.clj:816)"
 "pulse.render$render_pulse_card_for_display.invoke(render.clj:812)"
 "api.pulse$fn__57638$fn__57639.invoke(pulse.clj:162)"
 "api.pulse$fn__57638.invokeStatic(pulse.clj:162)"
 "api.pulse$fn__57638.invoke(pulse.clj:155)"
 "middleware.auth$enforce_authentication$fn__59463.invoke(auth.clj:14)"
 "routes$fn__60988$fn__60989.doInvoke(routes.clj:113)"
 "middleware.exceptions$catch_uncaught_exceptions$fn__59554.invoke(exceptions.clj:104)"
 "middleware.exceptions$catch_api_exceptions$fn__59551.invoke(exceptions.clj:92)"
 "middleware.log$log_api_call$fn__60864$fn__60865.invoke(log.clj:73)"
 "middleware.log$log_api_call$fn__60864.invoke(log.clj:66)"
 "middleware.security$add_security_headers$fn__59517.invoke(security.clj:102)"
 "middleware.json$wrap_json_body$fn__60687.invoke(json.clj:61)"
 "middleware.json$wrap_streamed_json_response$fn__60705.invoke(json.clj:120)"
 "middleware.session$bind_current_user$fn__58204$fn__58205.invoke(session.clj:159)"
 "middleware.session$do_with_current_user.invokeStatic(session.clj:142)"
``` 

Error messages in English:
- "Echec du réglage du fuseau horaire" => "Failed to set timezone"
- "Une erreur d'entrée/sortie a eu lieu lors d'envoi vers le serveur." => "An I/O error occurred when sending to the server"
- "Erreur au rendu de la carte de Pulse" => "Pulse card render error"

**Comment by jescorotero** [comment]
We have the same problem since v0.32

**Comment by pulponair** [comment]
Same issue:
https://github.com/metabase/metabase/issues/9989#issuecomment-498347396

This is all releated to the client not reconnecting if the database server closed the connection due to a timeout :(

**Comment by Folyd** [comment]
Same problem. Does it get fixed?

**Comment by cproctor1988** [comment]
Same issue on MariaDB, Any workarounds yet?
I have the same timeout issue on dashboards but they can be refreshed several times and it goes away. The scheduled Pulse can as well if you manually go in and refresh the pulse screen until it shows the results for all of your questions but it defeats the purpose of having the pulse .

**Comment by burgoon** [comment]
Still getting this on v0.33.3. Been getting since v0.28. Source data is in a MySQL DB, JAR is hosted on Amazon Linux EC2. I don't get the error in the dashboard, only in the Pulse.

**Comment by ShlomiCo** [comment]
Same issue on v0.33.3 when it's sent to slack

**Comment by flamber** [comment]
As noted previously, this is likely caused by Metabase not reconnecting to the database after timeout.
That problem was fixed in #11018 and will be part of 0.34.0, when it's released.

**Comment by mafouto** [comment]
We're still experiencing this issue on Redshift after updating to version 0.34.0.
This is a pain since we rely a lot on the pulses. 
Is there any workaround we could do?

**Comment by clkhub** [comment]
I've just upgraded to 0.34.2, but I'm still having the same issue. 
Redshift <--> Metabase
"An error occured while displaying this card" in Pulses

**Comment by flamber** [comment]
@clkhub Can you post the error from the log too? Admin > Troubleshooting > Logs

**Comment by mathiasrw** [comment]
Still getting same error. Any news on this?

**Comment by flamber** [comment]
@mathiasrw 
Can you post the error from the log? Admin > Troubleshooting > Logs
And please post "Diagnostic Info" from Admin > Troubleshooting too.

**Comment by mathiasrw** [comment]
@flamber sure!

Diagnostics:
```
{
  "browser-info": {
    "language": "da",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_222-heroku-b10",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_222-heroku",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.222-b10",
    "os.name": "Linux",
    "os.version": "4.4.0-1062-aws",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "mysql"
    ],
    "hosting-env": "heroku",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "tag": "v0.33.3",
      "date": "2019-09-20",
      "branch": "release-0.33.x",
      "hash": "86d32bf"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```


[metabaseLog.txt](https://github.com/metabase/metabase/files/4364494/metabaseLog.txt)


Aw. Lots of stuff in the log. We send out emails 7 am - but the log does not cover that period. Ill send the log next time we have the problem too.

**Comment by flamber** [comment]
@mathiasrw :wave: :denmark: The latest version is currently 0.34.3 - please try with that version. And if possible, please try with Java 11 - some pulse errors only happens on Java 8.

**Comment by mathiasrw** [comment]
Interesting. Ill see how I can upgrade the thing on heruko.

**Comment by flamber** [comment]
https://github.com/metabase/metabase/blob/master/docs/operations-guide/running-metabase-on-heroku.md#pinning-metabase-versions
Please use the forum for questions and troubleshooting: https://discourse.metabase.com/

**Comment by mafouto** [comment]
So, we're in version 0.35.0 and are still getting the error.
I can't find anything in the logs, but I found one message looking at the tasks area.

<img width="1044" alt="Screenshot 2020-03-24 at 09 27 01" src="https://user-images.githubusercontent.com/30630335/78238616-10ff9080-74dd-11ea-9466-aaf3dd2a16d9.png">

<img width="636" alt="Screenshot 2020-03-24 at 07 36 32" src="https://user-images.githubusercontent.com/30630335/78238577-093fec00-74dd-11ea-9c17-7b3d033e2cbc.png">

This is our diagnostics
```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.162 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "4.15.0-91-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "googleanalytics",
      "postgres",
      "redshift"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "5.7.26-log"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-03-31",
      "tag": "v0.35.0",
      "branch": "master",
      "hash": "3f96c47"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```

**Comment by flamber** [comment]
@mafouto There's a bug in 0.35.0, so stacktrace are not displayed in the log - #12233
Interesting that the task says it almost took 1.5 hours to complete the pulse (with failure).
Has this pulse ever worked? If yes, on which version and how long does it normally take?
Can you give more details about the questions included in the pulse?
There's a few of pulse issues open, but it has been really difficult to reproduce and fix them, so any help we can get to narrow down the cause and how to reproduce would be much appreciated.

**Comment by mathiasrw** [comment]
Its working on mine after updating to latest version.

**Comment by mafouto** [comment]
Hey, @flamber thanks for the fast reply :D 
First of all, apologies for the old screenshots, those were from the previous version. I've been taking screenshots daily and got them mixed up when posting.
This is one from today:
<img width="1006" alt="Screenshot 2020-04-03 at 07 39 47" src="https://user-images.githubusercontent.com/30630335/78330805-16b0b100-7585-11ea-83ef-1a55b6f27b28.png">
I also got an error message from one of the pulses in the log, but so far I didn't get the errors in the slack channel yet:
<img width="1047" alt="Screenshot 2020-04-03 at 07 41 28" src="https://user-images.githubusercontent.com/30630335/78330881-38119d00-7585-11ea-90d7-a9fb4bc96c00.png">


The issue is still happening, but I don't see the null pointer exception anymore when I click details, only a `null` message.
As far as I remember, we didn't have this issue until we updated to version 0.32.7 (this version is the one that had the upgrade to the latest Redshift JDBC driver version). I think we kept an older version for a while (like we jumped from 0.31 something to 0.32.7).
The pulses actually do work every now and then. Our pulses are mostly simple questions that don't normally take long to run even on dashboards. Also, when the pulses are sent as expected, they take only a few seconds - same if we trigger them manually.
And this behavior happens for almost all of our scheduled pulses. A few weeks back we found out that if we restarted Metabase during the night, the pulses worked the next morning. It worked for a while and then it went back to the same behavior again, with the error messages being displayed almost daily.
Another thing, I already checked the Java version, we have the 11, so it's not that either.
